app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/ToolbarUiController.kt

```diff
@@ -121,7 +127,11 @@ class ToolbarUiController(
         setOnFocusChangeListener { _, hasFocus -> if (!hasFocus) hasUserChangedURLSinceEditTextFocused = false }
     }
 
-    private fun initToolbar(layout: View, viewLifecycleOwner: LifecycleOwner, fragmentManager: FragmentManager) {
+    fun observeToolbarState(
+        layout: View,
+        viewLifecycleOwner: LifecycleOwner,
```

Comment:
I think we're creating multiple observers to `toolbarViewModel.legacyState`: I think you should separate this back into `initToolbar` and `observeToolbarState`, unless you plan to convert it all to Rx.